pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -542,10 +566,24 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 		return changes
 	}
 
+	//check sidecars status
+	sidecarStatus := status.GetSidecarsStatus(pod)
```

Comment:
shall the source of truth about pod status of this function `podStatus` instead of `pod.status`?

Reply:
The reason why `pod.Status` is used over `podStatus` is because `pod.Status` contains the info in regards to whether the sidecars are ready or not. `podStatus` can only tell us if they're running.